ext/hal/nordic/nrfx_glue.h

```diff
@@ -233,16 +233,39 @@ extern "C" {
 //------------------------------------------------------------------------------
 
 /** @brief Bitmask that defines DPPI channels that are reserved for use outside of the nrfx library. */
-#define NRFX_DPPI_CHANNELS_USED  0
+#define NRFX_DPPI_CHANNELS_USED 0
 
 /** @brief Bitmask that defines DPPI groups that are reserved for use outside of the nrfx library. */
-#define NRFX_DPPI_GROUPS_USED    0
+#define NRFX_DPPI_GROUPS_USED   0
 
 /** @brief Bitmask that defines PPI channels that are reserved for use outside of the nrfx library. */
-#define NRFX_PPI_CHANNELS_USED  0
+#define NRFX_PPI_CHANNELS_USED  (NRFX_PPI_CHANNELS_USED_BY_BT_CTLR | \
```

Comment:
nrfx_glue.h seems to be only included from your own nrfx hal top header, which is not part of the nrf52_bsim hal. So this change will not be visible there.

Reply:
Yes, I noticed. And I'm wondering how to solve this appropriately. Do you think that something like:
```
#if defined(SOC_SERIES_BSIM_NRFXX)
#define NRFX_PPI_CHANNELS_USED_BY_PWM_SW  0
#endif
```
could go? Perhaps with a comment that nrfx_glue.h is not processed when building for the simulator.

Reply:
where would you place those lines?

Reply:
Right before the assertion, [here](https://github.com/zephyrproject-rtos/zephyr/pull/15684/files#diff-d68c1b8276cada6142f4d495d302bea4R475).

Reply:
Good enough for me